apps_product_creator/models/product_product.py

```diff
@@ -0,0 +1,258 @@
+                    product.odoo_module_id.module_version_ids, attribute
+                )
+                for version in versions:
+                    module = version.module_id
+                    mod_ver_ids = module.dependence_module_version_ids.ids
+                    dependency_modules = module_version.search(
+                        [
+                            ("id", "in", mod_ver_ids),
+                            (
+                                "repository_branch_id",
+                                "=",
+                                version.repository_branch_id.id,
+                            ),
+                        ]
+                    )
+                    product_ids = []
+                    for dep in dependency_modules:
+                        product_variant_data = product_variant.search(
+                            [
+                                ("name", "=", dep.name),
+                                (
+                                    "odoo_module_version_id.module_id",
+                                    "=",
+                                    dep.module_id.id,
+                                ),
+                                (
+                                    "attribute_value_ids.name",
+                                    "=",
+                                    dep.repository_branch_id.name,
+                                ),
+                            ]
+                        )
+                        if product_variant_data:
+                            for pro in product_variant_data:
+                                product_ids.append(pro.id)
+                        if not product_variant_data:
+                            prd_data = dep.module_id._create_product()
+                            product_variant_data = prd_data.product_variant_ids.search(
+                                [
+                                    (
+                                        "attribute_value_ids.name",
+                                        "=",
+                                        dep.repository_branch_id.name,
+                                    ),
+                                    (
+                                        "odoo_module_version_id.module_id",
+                                        "=",
+                                        dep.module_id.id,
+                                    ),
+                                ]
+                            )
+                            if product_variant_data:
+                                product_ids.append(product_variant_data.id)
+                            values.update(
+                                {
+                                    "dependent_product_ids": [(6, 0, product_ids)]
+                                    if product_ids
+                                    else False,
+                                }
+                            )
+                        else:
```

Comment:
Move this else-block inside of the previous "if product_variant_data" (line 168)